pandas/tests/reshape/test_pivot.py

```diff
@@ -283,6 +283,23 @@ def test_pivot_multi_functions(self):
         expected = concat([means, stds], keys=['mean', 'std'], axis=1)
         tm.assert_frame_equal(result, expected)
 
+    # adding the test case for multiple columns as index (#21425)
+    def test_pivot_multiple_columns_as_index(self):
+        df = DataFrame({'lev1': [1, 1, 1, 1, 2, 2, 2, 2],
+                        'lev2': [1, 1, 2, 2, 1, 1, 2, 2],
+                        'lev3': [1, 2, 1, 2, 1, 2, 1, 2],
+                        'values': [0, 1, 2, 3, 4, 5, 6, 7]})
+        result = df.pivot(index=['lev1', 'lev2'],
+                          columns='lev3',
+                          values='values')
+        exp_index = pd.MultiIndex.from_product([[1, 2], [1, 2]],
+                                               names=['lev1', 'lev2'])
+        exp_columns = Index([1, 2], name='lev3')
+        expected = DataFrame([[0, 1], [2, 3], [4, 5], [6, 7]],
+                             exp_index,
+                             exp_columns)
+        tm.assert_frame_equal(result, expected)
```

Comment:
Can you add an example where the `values` argument is not mentioned? eg `df.pivot(index=['lev1', 'lev2'], columns='lev3')` 
(I think this will not yet work, but should work)